components/03-modules/form/form--compact.hbs

```diff
@@ -0,0 +1,9 @@
+<form class="form-compact {{ class }}">
```

Comment:
It is complex variant, therefore, I've used  `form-compac` instead of `form form--compact`. 
Does class has to be related with BEM modifier? You can check out sass file for this component. There are a lot of class added under `form-compac`.